pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -79,7 +79,8 @@ func (builder *Dockerfile) Build(config *api.Config) (*api.Result, error) {
 		config.AssembleUser = "1001"
 	}
 	if !user.IsUserAllowed(config.AssembleUser, &config.AllowedUIDs) {
-		return nil, s2ierr.NewUserNotAllowedError(config.AssembleUser, false)
+		builder.setFailureReason(utilstatus.ReasonAssembleUserForbidden, utilstatus.ReasonMessageAssembleUserForbidden)
+		return builder.result, s2ierr.NewUserNotAllowedError(config.AssembleUser, false)
```

Comment:
why couldn't (or why wasn't) the caller of Build() interpreting the returned error and handling this?  Seems a bit odd to do it everywhere we are about to return an error.

Reply:
the original s2i code relied on the returned `builder.result` to be non-nil. It would use this to record timings for the build.